### PR TITLE
Fix RedHat8 environment-modules config path and add test

### PR DESCRIPTION
**Pull request by francesco-giordano**
### Description of changes
* Add intel_mpi InSpec test 
* Fix RedHat8 environment-modules config path since the version of environment-modules is 4.x.x while in previous os was 3.x.x

### Tests
* Tested on docker and EC2

### References
* https://modules.readthedocs.io/en/v4.1.4/diff_v3_v4.html

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb [hunk 1]
```diff
@@ -43,7 +49,7 @@
   creates intelmpi_installation_path.to_s
 end
 
-append_if_no_line "append intel modules file dir to modules conf" do
+append_if_no_line "append intel modules file dir to modules conf #{node['cluster']['modulepath_config_file']}" do
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb:52]: The fact that the path changes in environment-modules 4.x let me think that if there is an update of this package in other OSes we'll be broken for that os as well.

I'm thinking at least in our kitchen spec tests we can have a  test to verify environment-modules version in every OS, to be sure to fail if the package is updated.

File: kitchen.validate.yml [hunk 2]
```diff
@@ -40,7 +40,6 @@ suites:
         - write_common_udev_configuration_files
         - ec2blkdev_service_installation
         - debian_udevd_reload_configuration
-        - package_repos
```
Comment (enrico-usai) [inline kitchen.validate.yml]: Why did you remove it? 

Is it needed only for intel_mpi? If this is not the case I prefer having it explicitly called.

**Comment by enrico-usai** [comment]
Can you please add more information in the commit messages? 
It's important to clarify this difference between 3.x and 4.x version in the commit message.

File: cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb (hunk 1 above)
Comment (francesco-giordano) [reply cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb:52]: I did not noticed but also ubuntu uses environment-modules version 4.x.x but they have solved the issue of compatibility adding directly in the package a link that point  `/etc/environment-modules/modulespath` to `/usr/share/modules/init/.modulespath`

There is already a test that verify that the intel_mpi module added in the cookbook is available

File: kitchen.validate.yml (hunk 2 above)
Comment (francesco-giordano) [reply kitchen.validate.yml]: No, I have removed it because that InSpec test has been remove in previous PR

**Review by enrico-usai — APPROVED** [review]
